File: Domain/Models/ProductModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain.Models
{
    public class ProductModel : Model
    {
        public long ProductId { get; set; }
        public string ProductName { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public bool Status { get; set; }
        public string PhotoUrl { get; set; }
        //public UserModel User { get; set; }
        public string UserId { get; set; }
    }

}

File: Web/ClientApp/src/app/Models/ProductModel.ts
export class ProductModel {
  price: number;
  productName: string;
  photoUrl: string;
  quantity: number;
  productId: number;
  userId: string;
}


export class Product {
  product: ProductModel;
  quantity: number;
}
export interface CartState {

  loaded: boolean;
  productOrder: ProductOrderModel[];

}
export interface ProductOrderModel {
  id: number;
  name: string;
  productId: number;
  quantity: number;
  price: number;
  added: boolean;
  buyerId: number;
  store: boolean;
  commission: number;
}
export interface ApiResponse<T> {
  statusCode?: number | undefined;
  message?: string | undefined;
  data?: T | undefined;
  errors?: string[] | undefined;
}
